__init__.py

```diff
@@ -54,99 +55,31 @@ def __init__(self):
         else:
             self.settings["max_volume"] = 100   # can be 0 to 100
         self.volume_sound = join(dirname(__file__), "blop-mark-diangelo.wav")
-        self.vol_before_mute = None
-        self._mixer = None
-
-    def _clear_mixer(self):
-        """For Unknown platforms reinstantiate the mixer.
-
-        For mycroft_mark_1 do not reinstantiate the mixer.
-        """
-        platform = self.config_core['enclosure'].get('platform', 'unknown')
-        if platform != 'mycroft_mark_1':
-            self._mixer = None
-
-    def _get_mixer(self):
-        self.log.debug('Finding Alsa Mixer for control...')
-        mixer = None
-        try:
-            # If there are only 1 mixer use that one
-            mixers = alsa_mixers()
-            if len(mixers) == 1:
-                mixer = Mixer(mixers[0])
-            elif 'Master' in mixers:
-                # Try using the default mixer (Master)
-                mixer = Mixer('Master')
-            elif 'PCM' in mixers:
-                # PCM is another common one
-                mixer = Mixer('PCM')
-            elif 'Digital' in mixers:
-                # My mixer is called 'Digital' (JustBoom DAC)
-                mixer = Mixer('Digital')
-            else:
-                # should be equivalent to 'Master'
-                mixer = Mixer()
-        except Exception:
-            # Retry instanciating the mixer with the built-in default
-            try:
-                mixer = Mixer()
-            except Exception as e:
-                self.log.error('Couldn\'t allocate mixer, {}'.format(repr(e)))
-        self._mixer = mixer
-        return mixer
+        # Instantiate the HAL emulator
+        self.platform = self.config_core['enclosure'].get('platform', 'unknown')
+        if self.platform in ALSA_PLATFORMS:
+            self.HAL = HALFactory.create('ALSA', self.settings)
```

Comment:
Pardon the intrusion, especially for a kind of style comment, it's not very important, I'm kind of wondering why this doesn't read

    `self.HAL = HALFactory.create(self.platform, self.settings)`

And let "create" decide what to return rather than needing to switch on platform here (given that it's a HAL).

Reply:
Or even `self.HAL = HALFactory.create(self.config_core, self.settings)`

Reply:
No intrusion at all - glad to have more eyes on it.

You're totally right here too - I'm thinking passing in the `self.platform` should be sufficient for now. 

I expect that this will change when it moves to the real HAL service, as each HAL plugin will either select their audio output service or provide their own methods that meet a minimum spec.